spec.md

```diff
@@ -601,10 +636,22 @@ message PluginCapability {
     Type type = 1;
   }
 
+  message AlphaFeature {
+    enum Type {
+      UNKNOWN = 0;
```

Comment:
Am I understand correctly that for each new alpha feature capability, a new enum will be added here? If that's the case, another option would be to mark the corresponding enum as alpha, like
```
message Service {
  enum Type {
    UNKNOWN = 0;
    CONTROLLER_SERVICE = 1;
    NEW_ALPHA_SERVICE=3 [(alpha_enum) = true];
  }
  Type type;
}
```

Reply:
you mean `alpha_enum_value` - right?

`PluginCapability` lets you pick a Service capability, or a VolumeExpansion capability. It's not clear to me whether all alpha features are a good fit for either. At the same time, PluginCapability.type is a `oneof` which has pretty strict rules re: backwards compat. If possible it would be nice to make that oneof strictly additive, which my mind ruled out alpha things. So I came up with AlphaFeature as a dumb bucket for things. And if we want to be consistent, people can (should?) just put their alpha features in there while they're iterating .. before things are promoted to stable.

I think that I also remember that someone objected to polluting Service/PluginCapability enums w/ alpha things - and AlphaFeature also seemed to check that box too.

That said, nothing stops someone from dumping an alpha-tagged enum value in the Service bucket. But I tried to avoid that in my examples (see #365) for the sake of consistency.

Also, it might be worth adding a NOTE to the oneof fields that they should not be polluted w/ alpha things. Docs...

Reply:
> At the same time, PluginCapability.type is a oneof which has pretty strict rules re: backwards compat. If possible it would be nice to make that oneof strictly additive, which my mind ruled out alpha things.

I think we can make it strictly additive, if an alpha feature does not make to the end, we can mark it as `deprecated`.

I think that bag of holding `AlphaFeatures` makes it hard to GA some fields/enums/... eventually as we'll have to remove it from `AlphaFeatures` and then add it to `Services` (or others).

Reply:
I think that was one of the objections raised, that people didn't want to carry around deprecated or outdated alpha API fields / types / values. we talked about maybe using "reserved" for things like this - as a way to declutter the API w/ respect to outdated names, but we'd still have the baggage of the old identifiers hanging around.

the way things have been going, i don't think the above concern is going to be a big problem and i'm happy to remove the AlphaFeature bag if others agree that this is the direction we should go.

Reply:
Done